src/main/java/bookstore/controller/BookController.java

```diff
@@ -36,18 +40,21 @@ public BookDto getBookById(@PathVariable Long id) {
 
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
-    public BookDto createBook(@RequestBody CreateBookRequestDto book) {
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto book) {
         return bookService.save(book);
     }
 
     @PutMapping("/{id}")
     @ResponseStatus(HttpStatus.OK)
-    public BookDto updateBookById(@PathVariable Long id, @RequestBody CreateBookRequestDto book) {
+    public BookDto updateBookById(
+            @PathVariable @Positive Long id,
+            @Valid @RequestBody CreateBookRequestDto book
+    ) {
         return bookService.updateById(id, book);
     }
 
     @DeleteMapping("/{id}")
-    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @ResponseStatus(HttpStatus.NOT_FOUND)
     public void deleteBookById(@PathVariable Long id) {
         bookService.deleteById(id);
     }
```

Comment:
What about adding validation to public BookDto getBookById(@PathVariable @Positive Long id) {